File: mybin/wiz.rb
#!/usr/bin/env ruby
def traverse_dir(file_path)  
  if File.directory? file_path  
    Dir.foreach(file_path) do |file|  
      if file!="." and file!=".."  
        traverse_dir(file_path+"/"+file){|x| yield x}  
      end  
    end  
  else  
    yield  file_path  
  end  
end  
  
s  = %q-.-  
old_path=''
traverse_dir(s){|f|  
 
# puts f  
 m=f.scan(/[^\/]*/)
 path=f.match(/.*(?=\/)/)
 #puts path
 if (path.to_s!=old_path.to_s)
   old_path=path
   tags=m[m.length-4]
  # puts tags
   content=m[m.length-2]
   # system "s=`cat \"#{f}\"`; echo $s; echo end"
#   system "~/@Work/Development/geeknote/gnsync.py --path \"#{path}\"  --notebook \"#{tags}\""
   print m
 end

} 

File: mybin/wiz.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import os 
def Test1(rootDir): 
	list_dirs = os.walk(rootDir) 
	for root, dirs, files in list_dirs: 
		for d in dirs: 
			path = os.path.join(root, d) 
			notebook = os.path.basename(d)    
			os.system("~/@Work/Development/geeknote/gnsync.py --path \"%s\"  --notebook  \"%s\" "% (path,notebook) )
			# os.system("echo  \" %s \"    \" %s \" " (path,notebook) )
			# os.system("echo \"%s\" \" %s\""%(path,notebook))

		for f in files: 
			pass
#			print os.path.join(root, f) 



def main():
	Test1(".")


if __name__=="__main__":
	main()
	tmp='haha'
	os.system("echo %s" % (tmp) )

